distributed/scheduler.py

```diff
@@ -2126,6 +2126,18 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws.occupancy == 0:  # special case to use round-robin
+                    wp_vals = worker_pool.values()
```

Comment:
To ensure efficient use of `WorkerState` objects, would suggest creating an annotated variable. Cython will understand this is an extension type and then can use C attribute access and C function calls on, which greatly improves performance on the object 

```suggestion
                    wp_vals = worker_pool.values()
                    wp_i: WorkerState
```